src/components/ListItem.svelte

```diff
@@ -0,0 +1,75 @@
+<script type="ts">
+  import TimeDifference from './TimeDifference.svelte';
+  import { timeDifference } from '../utils';
+  import { fade, fly } from 'svelte/transition';
+  import type { TodoItem } from '../types/item';
+
+  export let item: TodoItem;
+</script>
+
+<style>
+  .entry {
+    display: flex;
+    user-select: none;
+  }
+
+  .entry-done,
+  .entry-fade {
+    opacity: 0.55;
+  }
+
+  .entry-done .label {
+    text-decoration: line-through;
+  }
+
+  .label-with-number {
+    max-width: 80vw;
+    display: inline-flex;
+  }
+
+  .nes-checkbox + span::before,
+  .nes-checkbox:checked + span::before {
+    top: 2px !important;
+  }
+
+  .timer-badges {
+    width: 100%;
+    margin-bottom: 1rem;
+  }
+</style>
+
+<label
+  in:fade={{ duration: 500 }}
+  out:fly={{ x: -500, duration: 500 }}
+  class="entry {item.done ? 'entry-done' : ''}
+    {item.highlighted ? 'entry-highlight' : item.highlighted === false ? 'entry-fade' : ''}">
```

Comment:
maybe this `fade` could be its own property

Reply:
I don't know how to make this work just yet.

I know the in directive takes an animation function which is what fade is but I don't know how to reference the node to pass into the function.

Reply:
Oh what I meant was to have another property for the class instead of 
`class:entry-fade={item.highlighted === false}`

it would have 
`class:entry-fade={item.fade}`   or something like that


Changing the animations on a condition is not needed and hard to do in any framework :D

Reply:
Are you sure we should do this?
It wouldn't take long but I don't think it's the right direction.

Having a property for both highlighted and faded create a conflict because we're saying they're allowed to be both faded and highlighted at the same time.

The idea behind it was that there are three states for the list/list items
- No highlight selected, represented by null
- One item highlighted, represented by true ('entry-highlight')
- All other items faded, represented by false ('entry-fade')

We would have to manage the fade state in TaskList.svelte which already looks nasty

<ListItem
  item={{ ...item,
    id: _index + 1,
    highlighted: highlightItemIndex === -1 ? null : _index == highlightItemIndex }} 
    **fade: highlightItemIndex === -1 ? null : _index != highlightItemIndex }}** 
/>

Reply:
> which already looks nasty

yes, thats why I wanted them to be refactored as well, the checks  for highlighingIndex / `highlighed` were needed before your PR.

Highlight was needed before to have a "not"-rule in css, but in the end also was for "still look the same, so that other items can fade"

Now, I think `highlighed` could be removed completely and only use `fade`, because in ListItem `highlighed` doesn't do anything (or I didn't saw it there) 

Once that is removed the only binding / property we need to fill is:

`fade: highlightItemIndex !== -1 && _index != highlightItemIndex`

which makes all of it a lot easier, in my opinion :)

What do you think about this change ?

Reply:
Sounds good, I'm glad we were able to figure this out.